### PR TITLE
Add more e2e tests

### PR DESCRIPTION
**Pull request by timebertt**
**How to categorize this PR?**

/area testing dev-productivity quality
/kind enhancement

**What this PR does / why we need it**:

This PR lays the foundation for extending our e2e test suite (executed on PRs and periodically on master) and adds e2e tests for hibernation.

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/5310
As listed in https://github.com/gardener/gardener/issues/5310, we later want to add several more tests, documentation for developers, restructure the `/test` package, etc.

**Special notes for your reviewer**:

/squash

**Release note**:

```feature developer
More e2e tests will be executed on your PRs and periodically on `master` now. You can also run them on your development machine using `make test-e2e-local`.
```

### CODE REVIEW

**Comment by rfranzke** [comment]
image

Great! Looking forward again :)

/cc @ialidzhikov

**Review by rfranzke — APPROVED** [review]
Love it, that's just really nice :) Now we can very easily add more such tests.
/lgtm

/invite @dguendisch 
Please check the changes in the test framework :)

File: test/framework/gardener_utils.go [hunk 1]
```diff
@@ -106,14 +106,6 @@ func (f *GardenerFramework) GetShootProject(ctx context.Context, shootNamespace
 
 // createShootResource creates a shoot from a shoot Object
 func (f *GardenerFramework) createShootResource(ctx context.Context, shoot *gardencorev1beta1.Shoot) (*gardencorev1beta1.Shoot, error) {
-	err := f.GetShoot(ctx, shoot)
-	if err == nil {
-		return shoot, apierrors.NewAlreadyExists(gardencorev1beta1.Resource("shoots"), shoot.Namespace+"/"+shoot.Name)
-	}
```
Comment (dguendisch) [inline test/framework/gardener_utils.go:112]: why is the check for existing shoot no longer needed?

Comment (timebertt) [reply test/framework/gardener_utils.go:112]: The `Create` call below will also return an `AlreadyExists` error, if the shoot exists. Hence, there should be no need to manually construct it.

**Review by dguendisch — APPROVED** [review]
/lgtm for testfwk related changes

**Pull request merged**